test/Dialect/FIRRTL/layers-basic.mlir

```diff
@@ -0,0 +1,180 @@
+// RUN: circt-opt %s
+
+firrtl.circuit "layers-basic" {
+
+  //===--------------------------------------------------------------------===//
+  // Props
+  //===--------------------------------------------------------------------===//
+
+  firrtl.layer @A bind {
+    firrtl.layer @B bind {
+    }
+  }
+
+  firrtl.layer @B bind {}
+
+  firrtl.module @Foo(out %o : !firrtl.probe<uint<1>, @A>) {
+    %c = firrtl.constant 0 : !firrtl.uint<1>
+    %r = firrtl.ref.send %c : !firrtl.uint<1>
+    %s = firrtl.ref.cast %r : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+    firrtl.ref.define %o, %s : !firrtl.probe<uint<1>, @A>    
+  }
```

Comment:
This seems wrong? Why is this able to cast  or define outside of a `layerblock @A`, i.e., this seems like it needs an ambient layer of `A` which it doesn't have?

The rule may be that the cast can go to any layer that is legal based on the ambient layers of the cast.

I'm also surprised that the define is allowed here for the same reasons.

Reply:
The idea is that it's okay to add a layer requirement to a probe. If a probe is valid without layer A, then it should still be valid if layer A is enabled.

Reply:
Yep! This is equivalent to saying that if a probe is conditionally defined, it is also fine to make it unconditionally defined. Equivalently, a `layerblock` can always be inlined into its parent.